plugins/woocommerce/src/StoreApi/Utilities/AgenticCheckoutUtils.php

```diff
@@ -0,0 +1,342 @@
+ */
+class AgenticCheckoutUtils {
+
+	/**
+	 * Get the shared parameters schema for checkout session requests.
+	 *
+	 * @return array Parameters array.
+	 */
+	public static function get_shared_params() {
+		return [
+			'items'               => [
+				'description' => __( 'Line items to add to the cart.', 'woocommerce' ),
+				'type'        => 'array',
+				'items'       => [
+					'type'       => 'object',
+					'properties' => [
+						'id'       => [
+							'description' => __( 'Product ID.', 'woocommerce' ),
+							'type'        => 'string',
+						],
+						'quantity' => [
+							'description' => __( 'Quantity.', 'woocommerce' ),
+							'type'        => 'integer',
+							'minimum'     => 1,
+						],
+					],
+					'required'   => [ 'id', 'quantity' ],
+				],
+			],
+			'buyer'               => [
+				'description' => __( 'Buyer information.', 'woocommerce' ),
+				'type'        => 'object',
+				'properties'  => [
+					'first_name'   => [
+						'description' => __( 'First name.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'last_name'    => [
+						'description' => __( 'Last name.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'email'        => [
+						'description' => __( 'Email address.', 'woocommerce' ),
+						'type'        => 'string',
+						'format'      => 'email',
+					],
+					'phone_number' => [
+						'description' => __( 'Phone number.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+				],
+			],
+			'fulfillment_address' => [
+				'description' => __( 'Fulfillment/shipping address.', 'woocommerce' ),
+				'type'        => 'object',
+				'properties'  => [
+					'name'        => [
+						'description' => __( 'Full name.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'line_one'    => [
+						'description' => __( 'Address line 1.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'line_two'    => [
+						'description' => __( 'Address line 2.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'city'        => [
+						'description' => __( 'City.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'state'       => [
+						'description' => __( 'State/province.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'country'     => [
+						'description' => __( 'Country code (ISO 3166-1 alpha-2).', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'postal_code' => [
+						'description' => __( 'Postal/ZIP code.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+				],
+				'required'    => [ 'line_one', 'city', 'country', 'postal_code' ],
+			],
+		];
+	}
+
+	/**
+	 * Add items to cart from request.
+	 *
+	 * @param array          $items Items array from request.
+	 * @param CartController $cart_controller Cart controller instance.
+	 * @return \WP_REST_Response|null Returns error response on failure, null on success.
```

Comment:
This seems like an odd pattern. Can we either:

1. Use try/catch and throw errors on failure
2. Return a `WP_Error` and check value using `is_wp_error`.

This should make it behave more in line with other endpoints. I'm fairly certain if you use `WP_Error`, API can handle converting the error object into a properly formatted response for you.

Reply:
The main concern here is to have the response in the expected format defined by the specs. I will need to check more on this.